### PR TITLE
Make UI work for event receiver subtypes

### PR DESCRIPTION
**Pull request by davepagurek**
- Uses a dropdown in the UI to toggle between event receivers
- Update router to use the base controller for event receiver subtypes

### CODE REVIEW

File: app/controllers/event_receivers_controller.rb [hunk 1]
```diff
@@ -72,6 +72,6 @@ def set_event_receiver
   def event_receiver_params
     params
       .require(:event_receiver)
-      .permit(:start_time, :interval, :job_type_id)
+      .permit(:type, :start_time, :interval, :job_type_id, :regex, :stream, :return_code)
```
Comment (AndrewMcBurney) [inline app/controllers/event_receivers_controller.rb]: 👍

File: app/javascript/packs/receiver.jsx [hunk 2]
```diff
@@ -15,44 +16,76 @@ export default class Receiver extends React.PureComponent {
...
-    this.delete = this.delete.bind(this);
+    this.state = this.nullState();
+
+    [
+      'updateType',
+      'updateStartTime',
+      'cancelUpdate',
+      'save',
+      'delete'
+    ].forEach(fn => this[fn] = this[fn].bind(this));
```
Comment (AndrewMcBurney) [inline app/javascript/packs/receiver.jsx]: not `.bind` 😭

Comment (davepagurek) [reply app/javascript/packs/receiver.jsx]: :')

Back in the early days of React this happened automatically, but when they piggybacked off of ES6 classes they had to keep the default behaviour of unbound methods. So now you have to bind a bunch of stuff yourself, rip

Comment (AndrewMcBurney) [reply app/javascript/packs/receiver.jsx]: I thought the point of the frameworks was to abstract away the bindings 😢 sad reacc only

**Pull request merged**